src/senaite/core/content/analysisprofile.py

```diff
@@ -295,14 +297,26 @@ def setServices(self, value):
                 raise TypeError(
                     "Expected object, uid or record, got %r" % type(v))
             records.append({"uid": uid, "hidden": hidden})
+
+        if keep_inactive:
+            # keep inactive services so they come up again when reactivated
+            uids = [record.get("uid") for record in records]
+            for record in self.getRawServices():
+                uid = record.get("uid")
+                if uid in uids:
+                    continue
+                obj = api.get_object(uid)
+                if not api.is_active(obj):
+                    records.append(record)
+
         mutator = self.mutator("services")
         mutator(self, records)
 
     # BBB: AT schema field property
     Service = Services = property(getServices, setServices)
 
     @security.protected(permissions.View)
-    def getServiceUIDs(self):
+    def getServicesUIDs(self):
```

Comment:
Can we stick here to the naming `getServiceUIDs` instead of `getServicesUIDs`?

#### Reasoning (according to ChatGPT;))

1. **Clarity**: The phrase "Service UIDs" is clear and suggests that it refers to the unique identifiers associated with each "Service." Using `Service` in the singular keeps the focus on the type of UIDs being retrieved rather than suggesting that there are "Services" with "UIDs."

2. **Consistency**: This aligns with common naming conventions where the subject of the method (e.g., `Service`) is singular, while the return type (e.g., `UIDs`) indicates a collection.

3. **Parallel to Standard Naming Conventions**: Naming methods like `getUserIDs` (rather than `getUsersIDs`) or `getItemNames` (instead of `getItemsNames`) is common and widely understood.

### Summary

**`getServiceUIDs`** is concise, clear, and consistent with common naming practices, making it the better choice.

Reply:
jeje, ok.... my ChatGPT confirm yours... ;)

----8<-------------------------

The best accessor name would be getServiceUIDs.

Explanation:

- Consistency: "ServiceUIDs" suggests you are returning multiple UIDs for the "Service" objects, and using the singular form "Service" keeps it cleaner and more intuitive.

- Clarity: Even though "Services" is plural, using "ServiceUIDs" makes it clear that you are getting a list of UIDs related to the service objects as a whole, without the repetition of plurals.

So, getServiceUIDs is clearer and more concise.

----8<-------------------------

Oh man, this AI is teaching us already

Reply:
Reverted with https://github.com/senaite/senaite.core/pull/2637/commits/d465041bc3fe6f6909afe4577e8c71016e4edb4f